mongodump/oplog_dump.go

```diff
@@ -104,23 +100,16 @@ func oplogDocumentFilter(in []byte) ([]byte, error) {
 		return nil, fmt.Errorf("cannot dump with oplog if admin.system.version is modified")
 	}
 
-	for i := range rawD {
-		if rawD[i].Name == "o" {
-			var rawO bson.RawD
-			err = bson.Unmarshal(rawD[i].Value.Data, &rawO)
-			for j := range rawO {
-				if rawO[j].Name == "renameCollection" {
-					return nil, fmt.Errorf("cannot dump with oplog while renames occur")
-				}
-			}
-		}
+	if nsD.O != nil && nsD.O.RenameCollection != nil {
+		return nil, fmt.Errorf("cannot dump with oplog while renames occur")
 	}
-	return bson.Marshal(rawD)
+
+	return append([]byte(nil), in...), nil
```

Comment:
Rather than relying on append for the copy, I'd prefer an explicit `make` followed by `copy`.

I'm kind of annoyed that we're copying here, because it's not clear why it needs to be a copy.  I've filed [TOOLS-2256](https://jira.mongodb.org/browse/TOOLS-2256) about it.